topdown/eval.go

```diff
@@ -1594,7 +1626,7 @@ func (e evalFunc) evalOneRule(iter unifyIterator, rule *ast.Rule, prev *ast.Term
 				}
 			}
 
-			prev = result
+			prev = result // sr: ???
```

Comment:
If this closure is invoked multiple times, we need to detect conflicts...

Example:

`f(_) = y { a := [1,2]; y = a[_] }`

Reply:
Ah, I didn't see how there was any iteration involved, but the closure can _invoked multiple times_ 💡 thanks!